src/MLJInterface.jl

```diff
@@ -0,0 +1,309 @@
+    outs = [
+        let (out, flag) = eval_tree_array(eq[i], Xnew_t, fitresult.options)
+            !flag && error("Detected a NaN in evaluating expression.")
+            out
+        end for (i, eq) in zip(best_idx, equations)
+    ]
+    return reduce(hcat, outs)
+end
+
+function get_equation_strings_for(::SRRegressor, trees, options, variable_names)
+    return (t -> string_tree(t, options; variable_names=variable_names)).(trees)
+end
+function get_equation_strings_for(::MultitargetSRRegressor, trees, options, variable_names)
+    return [
+        (t -> string_tree(t, options; variable_names=variable_names)).(ts) for ts in trees
+    ]
+end
+
+function choose_best(; trees, losses::Vector{L}, scores, complexities) where {L<:LOSS_TYPE}
+    # Same as in PySR:
+    # https://github.com/MilesCranmer/PySR/blob/e74b8ad46b163c799908b3aa4d851cf8457c79ef/pysr/sr.py#L2318-L2332
+    # threshold = 1.5 * minimum_loss
+    # Then, we get max score of those below the threshold.
+    threshold = 1.5 * minimum(losses)
+    return argmax([
+        (losses[i] <= threshold) ? scores[i] : typemin(L) for i in eachindex(losses)
+    ])
+end
+
+function dispatch_selection_for(m::SRRegressor, trees, losses, scores, complexities)
+    return m.selection_method(;
+        trees=trees, losses=losses, scores=scores, complexities=complexities
+    )::Integer
+end
+function dispatch_selection_for(
+    m::MultitargetSRRegressor, trees, losses, scores, complexities
+)
+    return [
+        m.selection_method(;
+            trees=trees[i], losses=losses[i], scores=scores[i], complexities=complexities[i]
+        )::Integer for i in eachindex(trees)
+    ]
+end
+
+MMI.metadata_pkg(
+    AbstractSRRegressor;
+    name="SymbolicRegression",
+    uuid="8254be44-1295-4e6a-a16d-46603ac705cb",
+    url="https://github.com/MilesCranmer/SymbolicRegression.jl",
+    julia=true,
+    license="Apache-2.0",
+    is_wrapper=false,
+)
+
+MMI.metadata_model(
+    SRRegressor;
+    input_scitype=Union{MMI.Table(MMI.Continuous),AbstractMatrix{<:MMI.Continuous}},
+    target_scitype=AbstractVector{<:MMI.Continuous},
+    supports_weights=true,
+    reports_feature_importances=false,
+    load_path="SymbolicRegression.MLJInterfaceModule.SRRegressor",
+    human_name="Symbolic Regression via Evolutionary Search",
+)
+MMI.metadata_model(
+    MultitargetSRRegressor;
+    input_scitype=Union{MMI.Table(MMI.Continuous),AbstractMatrix{<:MMI.Continuous}},
+    target_scitype=Union{MMI.Table(MMI.Continuous),AbstractMatrix{<:MMI.Continuous}},
+    supports_weights=true,
+    reports_feature_importances=false,
+    load_path="SymbolicRegression.MLJInterfaceModule.MultitargetSRRegressor",
+    human_name="Multi-Target Symbolic Regression via Evolutionary Search",
+)
+
+function tag_with_docstring(model_name::Symbol, description::String)
+    docstring = """$(MMI.doc_header(eval(model_name)))
+
+    # Arguments
+    """
+
+    # TODO: These ones are copied (or written) manually:
```

Comment:
We should be able to reuse some of the documentation from the main codebase. Right?

Reply:
Yeah, most things are copied automatically in `OPTION_DESCRIPTIONS`. The remaining ones written out here are copied manually because the `equation_search` method has several arguments which are different from `SRRegressor`. But there should be a way to slice it cleanly in the future.